src/libraries/Common/src/System/Reflection/TypeNameParser.Helpers.cs

```diff
@@ -8,6 +8,28 @@
 
 #nullable enable
 
+#if SYSTEM_PRIVATE_CORELIB
+namespace System.Reflection.Metadata
+{
+    internal struct TypeNameParseOptions
+    {
+        public TypeNameParseOptions() { }
+#pragma warning disable CA1822 // Mark members as static
+        // CoreLib does not enforce any limits
```

Comment:
Thanks for the comment on this; makes sense that we need the helper to pull this off more cleanly.